builtin_array.go

```diff
@@ -1224,6 +1224,169 @@ func (r *Runtime) arrayproto_flatMap(call FunctionCall) Value {
 	return a
 }
 
+func (r *Runtime) arrayproto_with(call FunctionCall) Value {
+	o := call.This.ToObject(r)
+	relativeIndex := call.Argument(0).ToInteger()
+	value := call.Argument(1)
+	length := toLength(o.self.getStr("length", nil))
+
+	actualIndex := int64(0)
+	if relativeIndex >= 0 {
+		actualIndex = relativeIndex
+	} else {
+		actualIndex = length + relativeIndex
+	}
+	if actualIndex >= length || actualIndex < 0 {
+		panic(r.newError(r.getRangeError(), "Invalid index %s", call.Argument(0).String()))
+	}
+	a := r.newArrayLength(length)
+
+	for k := int64(0); k < length; k++ {
```

Comment:
I guess that if you are using the optimization below I would prefer that it done here as well.

Reply:
I will optimize this method.